Fix TestWorkflowRetry flakey test

```
2020-09-22T20:15:08.1567699Z --- FAIL: TestCLIWithServerSuite (211.00s)
2020-09-22T20:15:08.1568639Z     --- FAIL: TestCLIWithServerSuite/TestWorkflowRetry (11.23s)
2020-09-22T20:15:08.1570802Z         cli_test.go:643: 
2020-09-22T20:15:08.1571195Z             	Error Trace:	cli_test.go:643
2020-09-22T20:15:08.1571600Z             	            				when.go:244
2020-09-22T20:15:08.1571958Z             	            				when.go:253
2020-09-22T20:15:08.1572319Z             	            				cli_test.go:642
2020-09-22T20:15:08.1572766Z             	Error:      	Received unexpected error:
2020-09-22T20:15:08.1573197Z             	            	exit status 1
2020-09-22T20:15:08.1573968Z             	Test:       	TestCLIWithServerSuite/TestWorkflowRetry
```

https://github.com/argoproj/argo/runs/1150921691?check_suite_focus=true
https://github.com/argoproj/argo/suites/1229456411/artifacts/18490770

--- a/workflow/util/util.go
+++ b/workflow/util/util.go
@@ -665,7 +665,19 @@ func RetryWorkflow(kubeClient kubernetes.Interface, hydrator hydrator.Interface,
 	default:
 		return nil, errors.Errorf(errors.CodeBadRequest, "workflow must be Failed/Error to retry")
 	}
+	var updated *wfv1.Workflow
+	err := wait.ExponentialBackoff(retry.DefaultRetry, func() (bool, error) {
+		var err error
+		updated, err = retryWorkflow(kubeClient, hydrator, wfClient, wf, restartSuccessful, nodeFieldSelector)
+		return err == nil, err
+	})
+	if err != nil {
+		return nil, err
+	}
+	return updated, err
+}
 
+func retryWorkflow(kubeClient kubernetes.Interface, hydrator hydrator.Interface, wfClient v1alpha1.WorkflowInterface, wf *wfv1.Workflow, restartSuccessful bool, nodeFieldSelector string) (*wfv1.Workflow, error) {
 	err := hydrator.Hydrate(wf)
 	if err != nil {
 		return nil, err

